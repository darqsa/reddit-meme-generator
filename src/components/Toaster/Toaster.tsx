import {
  CreateContainerStyled,
  CreateSpanStyled,
  DeleteContainerStyled,
  ImageStyled,
  SpanStyled,
} from "./ToasterStyled";

interface ToasterProps {
  status: "memeCreated" | "memeDeleted" | "none";
}

const Toaster = ({ status }: ToasterProps): JSX.Element => {
  return (
    <>
      {status === "memeDeleted" && (
        <DeleteContainerStyled>
          <ImageStyled src="img/toaster.png" alt="It's a happy frog" />
          <SpanStyled>Your meme has been removed succesfully!</SpanStyled>
        </DeleteContainerStyled>
      )}
      {status === "memeCreated" && (
        <CreateContainerStyled>
          <ImageStyled src="img/toaster.png" alt="It's a happy frog" />
          <CreateSpanStyled>
            Your meme has been created succesfully!
          </CreateSpanStyled>
        </CreateContainerStyled>
      )}
      {status === "none" && <></>}
    </>
  );
};
export default Toaster;
