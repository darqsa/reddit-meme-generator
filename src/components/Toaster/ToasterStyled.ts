import styled from "styled-components";

const DeleteContainerStyled = styled.div`
  @media screen and (min-width: 700px) {
    width: 500px;
  }
  align-items: center;
  display: flex;
  flex-direction: column;
  border-radius: 20px;
  background-color: white;
  border: 3px solid red;
  height: 300px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  justify-content: center;
`;
const ImageStyled = styled.img`
  height: 160px;
`;
const SpanStyled = styled.span`
  color: red;
  text-align: center;
  font-size: 25px;
  margin: 20px 0;
  font-weight: normal;
`;
const CreateContainerStyled = styled.div`
  @media screen and (min-width: 700px) {
    width: 500px;
  }
  align-items: center;
  display: flex;
  flex-direction: column;
  border-radius: 20px;
  background-color: white;
  border: 3px solid #49c100;
  height: 300px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  justify-content: center;
`;
const CreateSpanStyled = styled.span`
  color: #49c100;
  text-align: center;
  font-size: 25px;
  margin: 20px 5px;
  font-weight: normal;
`;
export {
  CreateSpanStyled,
  CreateContainerStyled,
  DeleteContainerStyled,
  ImageStyled,
  SpanStyled,
};
