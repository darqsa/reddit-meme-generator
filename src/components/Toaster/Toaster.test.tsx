import TestRenderer from "react-test-renderer";
import Toaster from "./Toaster";

describe("Given a Toaster component", () => {
  describe("When it's rendered and recieves 'memeCreated'", () => {
    test("Then it should show a Toaster component with the 'created' conditional render", () => {
      const create = "memeCreated";

      const expectedToaster = TestRenderer.create(<Toaster status={create} />);
      expect(expectedToaster).toMatchSnapshot();
    });
  });
  describe("When it's rendered and recieves 'memeDeleted'", () => {
    test("Then it should show a Toaster component with the 'deleted' conditional render", () => {
      const create = "memeDeleted";

      const expectedToaster = TestRenderer.create(<Toaster status={create} />);
      expect(expectedToaster).toMatchSnapshot();
    });
  });
  describe("When it's rendered and recieves 'none'", () => {
    test("Then it shouldn't show anything", () => {
      const none = "none";

      const expectedToaster = TestRenderer.create(<Toaster status={none} />);
      expect(expectedToaster).toMatchSnapshot();
    });
  });
});
