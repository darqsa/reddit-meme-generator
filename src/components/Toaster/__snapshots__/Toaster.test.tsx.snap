// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Toaster component When it's rendered and recieves 'memeCreated' Then it should show a Toaster component with the 'created' conditional render 1`] = `
<div
  className="sc-hKMtZM eBccJa"
>
  <img
    alt="It's a happy frog"
    className="sc-gsnTZi jfOgbZ"
    src="img/toaster.png"
  />
  <span
    className="sc-eCYdqJ bTZEsv"
  >
    Your meme has been created succesfully!
  </span>
</div>
`;

exports[`Given a Toaster component When it's rendered and recieves 'memeDeleted' Then it should show a Toaster component with the 'deleted' conditional render 1`] = `
<div
  className="sc-bczRLJ bXeUiX"
>
  <img
    alt="It's a happy frog"
    className="sc-gsnTZi jfOgbZ"
    src="img/toaster.png"
  />
  <span
    className="sc-dkzDqf hPBPtn"
  >
    Your meme has been removed succesfully!
  </span>
</div>
`;

exports[`Given a Toaster component When it's rendered and recieves 'none' Then it shouldn't show anything 1`] = `null`;
